src/imageops/colorops.rs

```diff
@@ -181,14 +179,45 @@ where
 pub trait ColorMap {
     /// The color type on which the map operates on
     type Color;
-    /// Returns the index of the closed match of `color`
+    /// Returns the index of the closest match of `color`
     /// in the color map.
     fn index_of(&self, color: &Self::Color) -> usize;
+    /// Looks up color by index in the color map.  If `idx` is out of range for the color map,
+    /// `None` is returned.
+    fn lookup(&self, idx: usize) -> Option<Self::Color>;
```

Comment:
Looks good, except for the breaking change here. Can we add some default implementation returning `None`?

Possibly by adding a `has_lookup(&self) -> bool { false }` default implemented method as well?

Reply:
See [RFC 1105](https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md#minor-change-adding-a-defaulted-item) for which changes are considered minor (i.e. don't require a major version bump).

Reply:
Thanks for the feedback, I meant to ask what is the best way to change APIs in the initial PR.  Thank you for the link to RFC 1105.